framework/src/engine/consensus/consensus.ts

```diff
@@ -569,6 +574,30 @@ export class Consensus {
 			this._metrics.finalizedHeight.set(this._chain.finalizedHeight);
 			this._metrics.maxHeightCertified.set(block.header.aggregateCommit.height);
 			this._metrics.maxHeightPrevoted.set(block.header.maxHeightPrevoted);
+
+			try {
+				// Save inclusion proof when keys are provided
+				if (
+					(this._systemConfig.keepInclusionProofsForHeights > 0 ||
+						this._systemConfig.keepInclusionProofsForHeights === -1) &&
+					this._inclusionProofKeys.length > 0
+				) {
+					this._logger.info(`Starting saving inclusion proof at height ${block.header.height}`);
+					const result = await this._abi.prove({
+						keys: this._inclusionProofKeys,
+						stateRoot: block.header.stateRoot as Buffer,
+					});
+
+					await this._chain.dataAccess.setInclusionProofs(result.proof, block.header.height);
```

Comment:
the function name indicates it is saving multiple "proofs" but the variable names and log messages indicate a single "proof". imo would be better to align the function name accordingly, same for other places (getter/setter)

Reply:
a Proof received from stateDB contains proof for multiple keys and hence the confusion, I will try to improve the naming

Reply:
I think this is still open but in case you feel it's not necessary feel free to resolve it.

Reply:
I will resolve it in the other PR #9198 as of now I will merge this one